### PR TITLE
feat(feature-flags): Improve feature flags snippets instructions

### PR DESCRIPTION
**Pull request by kefranabg**
## Description

🎁 

I'm not a react expert, please let me know if I should do any changes 👍 

## Problem

Solves [#73](https://github.com/PostHog/posthog/issues/7394)

## Changes

- `FeatureFlagInstructions` component added => responsible of displaying FF instructions for all available code languages supported by posthog
- Removed the current collapse panel for feature flags examples and replaced it with the `FeatureFlagInstructions` component
- Added missing code lang icons (Node, PHP, Ruby, Go) (I used arbitrary svg icons, please let me know if you want to suggest your own svgs)
- Added missing snippets per language
- Added proper link to redirect to the right documentation page

https://user-images.githubusercontent.com/9840435/177139138-8eb49f6f-3657-4eba-b379-7e65572e60c7.mp4

Figma: https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=5545%3A30469

## How did you test this code?

- Ran unit tests
- Ran e2e tests
- Added a e2e test to cover my changes
- Checked that every component that I updated were not used in another context that could break anything
- Manually tested my changes
- Tested that each feature flag instruction was matching the right code snippet and documentation link

### CODE REVIEW

File: frontend/src/lib/components/icons.tsx
Comment (kefranabg) [inline frontend/src/lib/components/icons.tsx:35]: As mentioned in the PR desc, I used arbitrary svg icons, please let me know if you want to suggest your own svgs

**Comment by neilkakkar** [comment]
This is great, thanks @kefranabg . Will defer to @clarkus about the icons, but the code looks good to me! You might just need to run the formatter^

**Comment by clarkus** [comment]
> As mentioned in the PR desc, I used arbitrary svg icons, please let me know if you want to suggest your own svgs

I don't think the icons are critical here. We can keep these icons, or remove them and rely on the names of each snippet type. Thanks for working on this!

**Comment by liyiy** [comment]
hey @kefranabg ! We'll also be adding more code snippet support for our multivariate feature flag instructions and also to experimentation. Let me know if this is something you'd like to work on! I'd love to get this PR merged in ASAP, but if you're busy I can take this over. And you'll definitely get contributor recognition 🤗

**Comment by kefranabg** [comment]
> hey @kefranabg ! We'll also be adding more code snippet support for our multivariate feature flag instructions and also to experimentation. Let me know if this is something you'd like to work on! I'd love to get this PR merged in ASAP, but if you're busy I can take this over. And you'll definitely get contributor recognition 🤗

Hi @liyiy! Sorry for the late reply. Sure! shouldn't take too long to improve for multivariante feature flags & experiments. Feel free to tag me on the related issue (if there is any yet), I'd be happy to help on that!

I just fixed the prettier format, should be ready to merge 👌

**Comment by liyiy** [comment]
> > hey @kefranabg ! We'll also be adding more code snippet support for our multivariate feature flag instructions and also to experimentation. Let me know if this is something you'd like to work on! I'd love to get this PR merged in ASAP, but if you're busy I can take this over. And you'll definitely get contributor recognition 🤗
> 
> Hi @liyiy! Sorry for the late reply. Sure! shouldn't take too long to improve for multivariante feature flags & experiments. Feel free to tag me on the related issue (if there is any yet), I'd be happy to help on that!
> 
> I just fixed the prettier format, should be ready to merge 👌

You might have to merge from master to get some of the remaining tests to pass since this PR's gotten a bit a stale 😅

**Comment by kefranabg** [comment]
@liyiy done. It seems a manual trigger for the GH workflows is required from your side :)

**Review by liyiy — APPROVED** [review]
Thanks for the contribution @kefranabg ! 

Here's the updated issue for additional snippet instructions followup: https://github.com/PostHog/posthog/issues/10936

**Comment by neilkakkar** [comment]
@Twixes I know you were looking into this a while back, but seems like test failures for contributor PRs are still happening :$ , and the cypress tests don't seem to have run. Any leads?

**Comment by timgl** [comment]
@jacobwgillespie looks like we're having issues with depot and contributor PRs. Would you mind having a look?

**Comment by jacobwgillespie** [comment]
Taking a look now...

**Comment by jacobwgillespie** [comment]
@timgl That workflow doesn't use Depot at all, we landed on not using Depot for PRs due to permissions issues on GitHub's end, they remove the ability to access secrets or issue OIDC tokens for `pull_request` events originating from forks. So the failing workflow here is the original one pre-Depot (Depot is used after merge).

The offending line is likely this: https://github.com/PostHog/posthog/blob/19b57a4942e33d8f89460c5a780f8867511267d3/.github/workflows/docker-image-test.yml#L68. It's trying to push the container, but the `secrets.GITHUB_TOKEN` is limited to read-only access by that same security feature since this is a `pull_request` build.

**Comment by Twixes** [comment]
Yeah, I'm working on fixing this over at #10982.

**Comment by liyiy** [comment]
hey @kefranabg we're running into some CI issues with contributor PRs https://github.com/PostHog/posthog/pull/10982

To prevent this PR from hanging out and becoming stale/forgotten, an alternative solution here is that I could open up the PR and we/you can cherry pick your commits over and I'll manually add you to our contributors page :)

**Comment by kefranabg** [comment]
> hey @kefranabg we're running into some CI issues with contributor PRs #10982
> 
> To prevent this PR from hanging out and becoming stale/forgotten, an alternative solution here is that I could open up the PR and we/you can cherry pick your commits over and I'll manually add you to our contributors page :)

Let's do that !

**Comment by EDsCODE** [comment]
Hi @kefranabg. I've merged this branch into #11100 and merged so we can close this now. Thank you for contributing!

**Comment by liyiy** [comment]
@kefranabg let me know in 1-2 days if you've gotten an automatic email for a merch voucher or not !

**Pull request closed without merging**
